REGRESSION (286320@main): [iPadOS] YouTube in Safari requires two taps to start playback

whsieh opened the pull request:
#### 859231d19f00572077958b0fa8ae90a10d9bafcd
<pre>
REGRESSION (286320@main): [iPadOS] YouTube in Safari requires two taps to start playback
<a href="https://bugs.webkit.org/show_bug.cgi?id=285633">https://bugs.webkit.org/show_bug.cgi?id=285633</a>
<a href="https://rdar.apple.com/140237391">rdar://140237391</a>

Reviewed by Abrar Rahman Protyasha.

Guard the changes in 286320@main, so that it&apos;s behind a site-specific quirk.

* Source/WebCore/page/ios/ContentChangeObserver.cpp:
(WebCore::ContentChangeObserver::didAddMouseMoveRelatedEventListener):

* LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover-quirk-expected.txt: Renamed from LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover-expected.txt.
* LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover-quirk.html: Renamed from LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover.html.

Keep this layout test passing by overriding the top document URL for the purposes of determining
active quirks. Additionally add a `-quirk` suffix to the name.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldTreatAddingMouseOutEventListenerAsContentChange const):
(WebCore::handleSteamQuirks):
(WebCore::Quirks::determineRelevantQuirks):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:

Remove an unused member variable.

* Source/WebCore/page/ios/ContentChangeObserver.cpp:
(WebCore::ContentChangeObserver::didAddMouseMoveRelatedEventListener):

Canonical link: <a href="https://commits.webkit.org/288627@main">https://commits.webkit.org/288627@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/541ec05b7eba1825ad07aed95fb7f4052dda4b2e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/83969 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3587 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38270 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89044 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/34977 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/86054 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/3678 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/11554 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/65308 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23147 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87015 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/2737 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76296 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45602 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2668 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/30519 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34026 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/73638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31277 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90420 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11234 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8129 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/73767 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11458 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72125 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/72983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18044 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17266 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/2572 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11186 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/16658 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11034 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/14510 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/12806 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover.html:
```diff
@@ -8,6 +8,7 @@
 jsTestIsAsync = true;
 expandedContainer = false;
 window.internals?.settings.setContentChangeObserverEnabled(true);
+window.internals?.setTopDocumentURLForQuirks("https://store.steampowered.com");
```
Should we reflect this change in the file name too? I don't know if that's the norm for quirky tests.

whsieh replied:
Makes sense! I'll add a `-quirk` suffix to the test.

aprotyas commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1864,6 +1871,17 @@ static void handleCBSSportsQuirks(QuirksData& quirksData, const URL& quirksURL,
     quirksData.shouldSynthesizeTouchEventsAfterNonSyntheticClickQuirk = true;
 }
 
+static void handleSteamQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
```
You can drop the `UNUSED_PARAM` and do unnamed args? Up to you though

```
static void handleSteamQuirks(QuirksData& quirksData, const URL& /* quirksURL */, const String& quirksDomainString, const URL& /* documentURL */)
```

whsieh replied:
I was matching the style of the surrounding code, but I do think your suggestion is a bit cleaner — I'll change to that.

Thanks for the review!

whsieh committed:
REGRESSION (286320@main): [iPadOS] YouTube in Safari requires two taps to start playback
https://bugs.webkit.org/show_bug.cgi?id=285633
rdar://140237391

Reviewed by Abrar Rahman Protyasha.

Guard the changes in 286320@main, so that it's behind a site-specific quirk.

* Source/WebCore/page/ios/ContentChangeObserver.cpp:
(WebCore::ContentChangeObserver::didAddMouseMoveRelatedEventListener):

* LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover-quirk-expected.txt: Renamed from LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover-expected.txt.
* LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover-quirk.html: Renamed from LayoutTests/fast/events/touch/ios/content-observation/add-mouseout-event-listener-on-hover.html.

Keep this layout test passing by overriding the top document URL for the purposes of determining
active quirks. Additionally add a `-quirk` suffix to the name.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldTreatAddingMouseOutEventListenerAsContentChange const):
(WebCore::handleSteamQuirks):
(WebCore::Quirks::determineRelevantQuirks):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:

Remove an unused member variable.

* Source/WebCore/page/ios/ContentChangeObserver.cpp:
(WebCore::ContentChangeObserver::didAddMouseMoveRelatedEventListener):

Canonical link: https://commits.webkit.org/288627@main
